helper/threads.c

```diff
@@ -407,6 +419,34 @@ int odph_thread_join(odph_thread_t thread[], int num)
 	return i;
 }
 
+int odph_thread_join_result(odph_thread_t thread[], odph_thread_join_result_t res[], int num)
+{
+	odph_thread_start_args_t *start_args;
+	int num_exited = 0;
+
+	for (int i = 0; i < num; i++) {
+		start_args = &thread[i].start_args;
+
+		if (start_args->status != STARTED) {
+			ODPH_DBG("Thread (i:%i) not started.\n", i);
+			continue;
+		}
+
+		if (thread[i].start_args.mem_model == ODP_MEM_MODEL_THREAD) {
+			if (wait_pthread(&thread[i], &res[i]))
+				return -1;
+		} else {
+			if (wait_process(&thread[i], &res[i]))
+				return -1;
```

Comment:
More informative for application to return `-1` only when no threads joined and otherwise `i`?